### PR TITLE
[upgrade/pack_builder] Make targetname size restriction explicit and check

### PR DESCRIPTION
**Pull request by oguzcanphilips**

### CODE REVIEW

File: upgrade/pack_builder/Input.hpp [hunk 1]
```diff
@@ -6,6 +6,15 @@
 
 namespace application
 {
+    struct TargetNameTooLongException
```
Comment (rjaegers) [inline upgrade/pack_builder/Input.hpp]: Why a struct? You could also hide the data members and implement the what() method.

File: upgrade/pack_builder/Input.hpp [hunk 2]
```diff
@@ -6,6 +6,15 @@
...
+        : std::exception
+    {
+        explicit TargetNameTooLongException(const std::string& name, int maxSize);
```
Comment (rjaegers) [inline upgrade/pack_builder/Input.hpp]: explicit not really necessary here.

File: upgrade/pack_builder/Input.hpp (hunk 1 above)
Comment (oguzcanphilips) [reply upgrade/pack_builder/Input.hpp]: All the other exception types used in UpgradePackBuilderFacade::Build were defined as such structs with public fields. Adapted them all.

File: upgrade/pack_builder/Input.hpp (hunk 2 above)
Comment (oguzcanphilips) [reply upgrade/pack_builder/Input.hpp]: Removed.

File: upgrade/pack_builder/SparseVector.hpp [hunk 3]
```diff
@@ -60,6 +66,13 @@ namespace application
         : position(position)
     {}
 
+    inline char const* OverwriteException::what() const
+    {
+        std::stringstream ss;
+        ss << "Contents specified twice for memory location at address 0x" << std::hex << std::setw(8) << std::setfill('0') << position;
+        return ss.str().c_str();
```
Comment (rjaegers) [inline upgrade/pack_builder/SparseVector.hpp]: You are returning a pointer to local storage here. Referring it later will lead to undefined behavior.

**Pull request merged**